tt_metal/hw/inc/dataflow_api.h

```diff
@@ -1575,3 +1585,108 @@ class Buffer {
         }
     }
 };
+
+FORCE_INLINE
+uint32_t min(uint32_t a, uint32_t b) { return (a < b) ? a: b; }
+
+#define NUM_ENTRIES_PER_SHARD 3
```

Comment:
where is this constraint coming from?

Reply:
This NUM_ENTRIES_PER_SHARD refers to how we populate the command_ptr on the host side. 
For each Shard we populate the number of  pages, the core's x coordinate and the core's y coordinate. 
I can add that as a comment.